subsys/bluetooth/mesh/access.c

```diff
@@ -1555,11 +1546,33 @@ int bt_mesh_model_recv(struct bt_mesh_msg_ctx *ctx, struct net_buf_simple *buf)
 		for (index = 0; index < dev_comp->elem_count; index++) {
 			const struct bt_mesh_elem *elem = &dev_comp->elem[index];
 
-			(void)element_model_recv(ctx, buf, elem, opcode);
+			int err_elem = element_model_recv(ctx, buf, elem, opcode);
+
+			err = err_elem != ACCESS_STATUS_SUCCESS ? err_elem : err;
```

Comment:
Changed comparison.  In your suggestions, it will always return ACCESS_STATUS_SUCCESS.

Reply:
I'm not following why. If at least one model was able to process the message, the message (seqnum) shall be stored in RPL (or am I'm missing something?). With this comparison, if at least one model returned error, `bt_mesh_model_recv` will return error and the message won't be stored in RPL.

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/70437#discussion_r1533598100
Hmm...  How it is going to work?
```
int err = ACCESS_STATUS_SUCCESS;
...
  		for (index = 0; index < dev_comp->elem_count; index++) {
 			const struct bt_mesh_elem *elem = &dev_comp->elem[index];
 
			int err_elem = element_model_recv(ctx, buf, elem, opcode);

			err = err_elem == ACCESS_STATUS_SUCCESS ? err_elem : err;
 		}
...
```
if it is `ACCESS_STATUS_SUCCESS` then return `ACCESS_STATUS_SUCCESS`. Otherwise `ACCESS_STATUS_SUCCESS`.
Any mistakes in my logic?

Reply:
Perhaps I incorrectly copy-pasted your code. Use the comparison from your code.